src/core_plugins/kibana/public/dashboard/dashboard_config.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Part of the exposed plugin API - do not remove without careful consideration.
+ * @type {boolean}
+ */
+let hideDashboardWriteControls = false;
```

Comment:
Cool, thanks for the detailed explanation and examples.  I was hoping to cut angular out of the equation but I suppose, until we have the new architecture, there is no other implementation (that I can think of anyway) that addresses the pro's you listed.  Well, I could write it so it's read only without angular, but I don't see how to do it in a way that ensure it's only modified before any application code runs.